app/src/main/java/com/vci/vectorcamapp/complete_session/details/presentation/components/specimens/CompleteSessionSpecimens.kt

```diff
@@ -31,25 +49,72 @@ fun CompleteSessionSpecimens(
             modifier = modifier.fillMaxSize()
         )
     } else {
-        LazyRow(
-            modifier = modifier.fillMaxSize(), horizontalArrangement = Arrangement.Center
+        val filteredSpecimenResults = specimensWithImagesAndInferenceResults.filter { specimenGroup ->
```

Comment:
Is there no way we can have this filtering logic be inside the view model without needing an additional list for filtered specimens?